### PR TITLE
Roll forward nonroot docker container

### PR DESCRIPTION
**Pull request by cgdolan**
PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by bmahe — APPROVED** [review]
lgtm.
How do we test it and prevent potential breakage?

**Comment by cgdolan** [comment]
> 

I'll run a release dry-run and verify the tags are correct on the images that get built.

edit: release dry-run looks good

root build not adding suffix to latest: https://github.com/returntocorp/semgrep/actions/runs/6091826334/job/16528987163#step:4:34
non-root build adding suffix to latest: https://github.com/returntocorp/semgrep/actions/runs/6091826334/job/16528986946#step:4:35

**Comment by aryx** [comment]
Are we sure we want to merge this? Wouldn't it be better to first put in place some testing mechanism to make sure we don't get the mess we got last week where customers could not run semgrep? We need some test infra in place first! I know this is tied to the release, so it's complicated to have such test infra in our CI, but maybe we need some canary style thing.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -327,6 +330,36 @@ jobs:
       repository-name: ${{ github.repository }}
       dry-run: false
 
+  build-test-docker-nonroot:
+    uses: ./.github/workflows/build-test-docker.yaml
+    secrets: inherit
+    with:
+      docker-flavor: |
+        latest=auto
```
Comment (aryx) [inline .github/workflows/tests.yml:338]: This requires a giant comment saying how important are those lines, and how we can get a big mess if we don't do the right thing here, with some links to some docs.

**Pull request merged**